### PR TITLE
Integrate gradle license plugin

### LINKED ISSUE #13: License update?
Do we need/want to update the license owner with NOTICE to original author?

Original intellij-lsp uses apache license (it's ok), but doesn't have copyright boilerplates in files (common mistake with apache :) )

So... at least we should add license header in all scala files with gradle license plugin (also i'll add it as part of CI process).
Also i think that java files from #4 can be considered as new code (based on original work), and we can add intellij-lsp-community copyright

@intellij-lsp/all?

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
Adds gradle license plugin and generated license headers.

Resolves #13

### CODE REVIEW

File: resources/HEADER [hunk 1]
```diff
@@ -0,0 +1,13 @@
+    Copyright 2017 Guillaume Tâche
```
Comment (nixel2007) [inline resources/HEADER]: I think we should mark it as 2017-2018.

**Comment by nixel2007** [comment]
Could you add a NOTICE file also? With something like "This software is based on intellij-lsp (link) by author-name (link) " and other words required by apache license?

File: build.gradle [hunk 2]
```diff
@@ -1,20 +1,23 @@
...
-apply plugin: 'org.jetbrains.intellij'
-
 intellij {
     version '2017.3'
     plugins 'IntelliLang'
     pluginName 'LSP Support'
 }
 
+license {
+    header rootProject.file('resources/HEADER')
```
Comment (nixel2007) [inline build.gradle:18]: Add this also pls to pass travis checks (can't push to your fork)

```groovy    
    strictCheck = true
    exclude "**/*.properties"
    exclude "**/*.xml"
```

Comment (NipunaRanasinghe) [reply build.gradle:18]: Added

**Comment by NipunaRanasinghe** [comment]
Codeclimate failures are occured due to a licence plugin issue. (https://github.com/hierynomus/license-gradle-plugin/issues/166)

**Comment by nixel2007** [comment]
You can config this behavior by `license.mapping` in gradle. will you give it a try?

**Comment by NipunaRanasinghe** [comment]
Sure will do. But IMO this should be fixed by the plugin itself. Anyway will configure for now.

**Comment by nixel2007** [comment]
LGTM!

I'll try to tweak codeclimate check.

**Review by nixel2007 — APPROVED** [review]

**Pull request merged**